user-interface/src/main/java/life/qbic/datamanager/views/projects/project/ontology/OntologyLookupComponent.java

```diff
@@ -0,0 +1,235 @@
+    searchField.setPrefixComponent(VaadinIcon.SEARCH.create());
+    searchField.setValueChangeMode(ValueChangeMode.LAZY);
+    searchField.addValueChangeListener(
+        event -> {
+          if (event.getValue().length() >= ONTOLOGY_SEARCH_LOWER_LIMIT) {
+            fireEvent(new OntologySearchEvent(this, event.isFromClient(), event.getValue()));
+          }
+        });
+    add(searchField);
+  }
+
+
+  /**
+   * Ontology Item
+   * <p>
+   * The ontology Item is a Div container styled similiar to the {@link Card} component, hosting the
+   * ontology information of interest provided by a {@link OntologyClassDTO}
+   */
+  private static class OntologyItem extends Div {
+
+    public OntologyItem(String label, String name, String classIri, String description,
+        String ontologyAbbreviation) {
+      addHeader(label, name);
+      addUrl(classIri);
+      addDescription(description);
+      addOrigin(ontologyAbbreviation);
+      addClassName("ontology-item");
+    }
+
+    private void addHeader(String label, String nameText) {
+      Span title = new Span(label);
+      title.addClassName("ontology-item-title");
+      Span name = new Tag(nameText);
+      name.addClassNames("primary", "clickable");
+      name.addClickListener(
+          event -> UI.getCurrent().getPage().executeJs("window.copyToClipboard($0)", nameText));
+      Span header = new Span(title, name, initCopyIcon(nameText));
+      header.addClassName("header");
+      add(header);
+    }
+
+    private void addUrl(String ontologyURL) {
+      Span url = new Span(ontologyURL);
+      url.addClassName("url");
+      add(url);
+    }
+
+    private void addDescription(String ontologyDescription) {
+      Div description = new Div();
+      description.add(ontologyDescription);
+      description.addClassName("description");
+      add(description);
+    }
+
+    private Icon initCopyIcon(String copyContent) {
+      Icon copyIcon = VaadinIcon.COPY_O.create();
+      copyIcon.addClassName(IconSize.SMALL);
+      copyIcon.addClassNames("clickable", "copy-icon");
+      copyIcon.addClickListener(
+          event -> UI.getCurrent().getPage().executeJs("window.copyToClipboard($0)", copyContent));
+      return copyIcon;
+    }
+
+    private void addOrigin(String origin) {
+      Tag originAbbreviation = new Tag(origin);
+      originAbbreviation.addClassName("primary");
+      Span originPreText = new Span("Comes from the Ontology: ");
+      originPreText.addClassName("secondary");
+      Span ontologyOrigin = new Span(originPreText, originAbbreviation);
+      ontologyOrigin.addClassName("origin");
+      add(ontologyOrigin);
+    }
+  }
+
+  /**
+   * <b>Ontology Search Event</b>
+   *
+   * <p>Indicates that an user wants to search for a list of {@link OntologyClassDTO} containing
+   * the provided value within the {@link OntologyLookupComponent}</p>
+   */
+  public static class OntologySearchEvent extends ComponentEvent<OntologyLookupComponent> {
```

Comment:
The event was just introduced to make the event propagation more clear and distinct since we currently expect at least 2 values provided in the searchfield.